packages/engine/tests/simulations/mod.rs

```diff
@@ -72,7 +72,28 @@ macro_rules! run_test {
 }
 
 mod state {
-    run_test!(edit_state_number);
+    run_test!(state_edit_bool_dot);
```

Comment:
I'm wondering if we should introduce a structure within the tests/simulations folder. Having them all flat in the directory makes it a bit hard to visually grok what we're testing, can we perhaps move them into folders like `state`, `globals` etc. and then perhaps even sub-folders within that if there are clear groupings?

Reply:
I have also thought about that as I implemented a bunch of other tests for the other areas. I think we could simply put (Rust) modules inside of `tests/simulations/` and call the macro there.

Reply:
I don't quite understand what you're suggesting, could you give an example of what the directory structure would look like?

Reply:
Sure, for example for states (this PR):
- tests
  - mod.rs (1)
  - simulations
    - mod.rs (2)
    - state
      - mod.rs (3) 
      - edit_bool_dot
      - edit_number_dit
    - ...
    
(1): `mod simulations;`
(2): `mod state; mod global; ...`
(3): `run_test!(edit_bool_dot); run_test!(...);`

Reply:
I'm a fan of that, perhaps we can come up with a better name for `simulations`* but in the mean-time I think that'd be a good improvement.

* To describe our philosophy around more 'unit-test' style integration testing

Reply:
I'd put this in a separate pull request so the other PRs I have opened (and will open in the near future) can use this change.

Reply:
I'd apply the suggested change before merging this PR or any other integration tests PR